internal/host/validator.go

```diff
@@ -1680,3 +1680,59 @@ func (v *validator) noSkipMissingDisk(c *validationContext) (ValidationStatus, s
 	}
 	return ValidationSuccess, successMessage
 }
+
+func (v *validator) noIPCollisions(c *validationContext) (ValidationStatus, string) {
+	if c.cluster != nil && common.IsDay2Cluster(c.cluster) {
+		return ValidationSuccessSuppressOutput, fmt.Sprintf("Skipping validation for day 2 host %s", c.host.ID)
+	}
+	if len(c.host.Connectivity) == 0 {
+		return ValidationPending, fmt.Sprintf("Awaiting connectivity report for host %s", c.host.ID)
+	}
+	connectivityReport, err := hostutil.UnmarshalConnectivityReport(c.host.Connectivity)
+	if err != nil {
+		message := fmt.Sprintf("Unable to unmarshall host connectivity for %s: %s", c.host.ID, err)
+		v.log.Errorf(message)
+		return ValidationError, message
+	}
+
+	collidingIPSWithMacs := getCollidingIPs(connectivityReport)
+	if len(collidingIPSWithMacs) > 0 {
+		message := "There was an IP collision detected.\n"
+		if len(collidingIPSWithMacs) > 1 {
+			message = "There were multiple IP collissions detected.\n"
+		}
+		for ipAddress, macInfo := range collidingIPSWithMacs {
+			message += fmt.Sprintf("Host %s detected collisions for IP %s, the following macs are in collision (%s)\n", c.host.ID, ipAddress, strings.Join(macInfo, ","))
+
+		}
+		return ValidationFailure, message
```

Comment:
Is it really a failure when we notice an IP collision on a network interface that is not part of the cluster's machine network? I wouldn't really care for collisions on something like that, as it will not be used for cluster-critical networking

Reply:
I cannot think of a customer that would have a topology that has duplicated IP addresses in a single IP segment and be still a valid network topology. Of course I agree that if this happens outside of the Machine Network subnet we will most probably never notice this, but nevertheless if our goal is to assist users, I would vote that this is an issue user should be told about.

I do not push for any direction, but from what we observe super-power-users run Podman on-prem anyway so it does not matter much. For a regular "please hold my hand" type of a user I believe such a message may help him in a long run (not necessarily just for the installation process)

Reply:
>I would vote that this is an issue user should be told about

Yes, but in this case we block the user from installing the cluster, we're not just telling them about it.

But you're correct, making this machine network specific is probably not worth it

Reply:
I didn't want to be as direct, but I meant to say "regular user will only thank us for blocking him; power user will need a few seconds to disable this validation". I know that blocking the installation is a fragile subject, but I personally think that we should stop users from using infrastructure that is wrong in some fundamental ways, even if those problems do not impact the installation process itself